scripts/tests/sanitycheck/test_reporting_testsuite.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''
+This test file contains testcases for reporting functionality of Testsuite class of sanitycheck
+'''
+import sys
+import os
+import xml.etree.ElementTree as ET
+import pytest
+import pandas as pd
+from mock import MagicMock
+
+ZEPHYR_BASE = os.getenv("ZEPHYR_BASE")
+sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts/sanity_chk"))
+
+
+def test_discard_report(class_testsuite, platforms_list, all_testcases_dict, caplog, tmpdir):
+    """ Testing discard_report function of Testsuite class in sanitycheck
+	Test 1: Check if apply_filters function has been run before running
+	discard_report
+	Test 2: Test if the generated report is not empty
+	Test 3: Test if the gerenrated report contains the expected columns"""
+    class_testsuite.platforms = platforms_list
+    class_testsuite.testcases = all_testcases_dict
+    filename = tmpdir.mkdir("test_discard").join("discard_report.csv")
+    with pytest.raises(SystemExit):
+        class_testsuite.discard_report(filename)
+    assert "apply_filters() hasn't been run!" in caplog.text
+
+    kwargs = {"exclude_tag" : ['test_a'], "exclude_platform" : ['demo_board_1'],
+              "platform" : ['demo_board_2']}
+    class_testsuite.apply_filters(**kwargs)
+    class_testsuite.discard_report(filename)
+    discard_report_df = pd.read_csv(filename)
```

Comment:
Do we really need this dependency just to read a csv file? Why dont use just use the standard csv module?

Reply:
We can use csv module to read the csv file, but did not find many easily provided functions in csv module like empty which pandas provide.

Reply:
csv module returns a list, if that list is empty, then you have no entries, no need to use pandas or whatever for something that is as simple.

Reply:
I am checking that the list is not empty for this case and then comparing the column values to expected values, which is why I used pandas to remove the NA's.

Reply:
Worked around the use of pandas library by using csv module. Not removing NA's now but dealing with None's in expected list of column values.